core/templates/dev/head/components/StoryCreationService.js

```diff
@@ -59,22 +59,28 @@ oppia.factory('StoryCreationService', [
           }
           storyCreationInProgress = true;
           AlertsService.clearWarnings();
-
+          var topic = TopicEditorStateService.getTopic();
           $rootScope.loadingMessage = 'Creating story';
           var createStoryUrl = UrlInterpolationService.interpolateUrl(
             STORY_CREATOR_URL_TEMPLATE, {
-              topic_id: topicId
+              topic_id: topic.getId()
             }
           );
           $http.post(createStoryUrl, {title: storyTitle})
             .then(function(response) {
               $timeout(function() {
-                $window.location = UrlInterpolationService.interpolateUrl(
-                  STORY_EDITOR_URL_TEMPLATE, {
-                    topic_id: topicId,
-                    story_id: response.data.storyId
-                  }
-                );
+                TopicUpdateService.addCanonicalStoryId(
```

Comment:
What if the user creates a story and then discards changes in the topic?

Would it be better to say "no creating/deleting stories until you've saved pending changes to the topic", and also treat creation/deletion of stories as atomic changelists (i.e. such an action results in a changelist immediately being saved with just that command)?

Reply:
The user is given a prompt if there are any pending changes left before creating the story. Once, create story is clicked, and the title is entered, the user is transferred to the story editor. So, for the user, Create Story would not need a 'Save Draft' to take effect, since they are directed to the story editor as soon as Create Story is clicked.

Reply:
But transferring the user to the story editor is not enough. Wouldn't the "story is added to topic" action also need to be committed? Otherwise you run into the scenario above where the user can then discard pending changes to the topic and you have an orphan story.

Reply:
Replied above for this also, but yeah, the add story change is immediately saved, before redirection to story editor.